### ISSUE TITLE
migrate fails on documents.0032_auto_20160315_0537

### ISSUE THREAD

**Issue by simeon-walker** [mayan-edms/mayan-edms#251]
In Oracle this column type change cannot happen unless it is empty. Is it an essential change?

```
  Applying documents.0032_auto_20160315_0537...Traceback (most recent call last):
  File "manage.py", line 9, in <module>
    execute_from_command_line(sys.argv)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/migrations/operations/fields.py", line 201, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/oracle/schema.py", line 53, in alter_field
    super(DatabaseSchemaEditor, self).alter_field(model, old_field, new_field, strict)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 484, in alter_field
    old_db_params, new_db_params, strict)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 636, in _alter_field
    params,
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/base/schema.py", line 111, in execute
    cursor.execute(sql, params)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/vagrant/Projects/edms_project/trunk/virtenv/lib/python2.7/site-packages/django/db/backends/oracle/base.py", line 478, in execute
    return self.cursor.execute(query, self._param_generator(params))
django.db.utils.DatabaseError: ORA-01439: column to be modified must be empty to change datatype
```

**Comment by rosarior** [comment]
This migration is important (https://github.com/mayan-edms/mayan-edms/blob/master/mayan/apps/documents/migrations/0032_auto_20160315_0537.py). Document UUIDs are used in several places, leaving this field blank will affect functionality. 

This migration moves the UUID field from a simple Char field to use Django's new specialized UUID Field class. A similar issue happened with Postgresql. 

```
 django.db.utils.ProgrammingError: column "uuid" cannot be cast automatically to type uuid
```

The solution was to use a custom SQL statement: https://github.com/mayan-edms/mayan-edms/commit/bd648d0cb19d47bc03c3475540087f1f38874da5#diff-fc3c565a9a99e5c1c67a0c64d11f58e6

It seems a similar approach will be needed for Oracle.

**Comment by simeon-walker** [comment]
Oracle doesn't have a UUID column type and the Django backend uses varchar2(32) for this. Unfortunately the Document model has already caused this field to be created nvarchar2(48) which can't easily be migrated to varchar2. The system may work just fine without this migration.

**Comment by rosarior** [comment]
Any problems disabling this migration so far? If none, the solution for the cabinet's select_for_update can be applied here and disable this migration for Oracle.

**Comment by simeon-walker** [comment]
No, haven't seen any issues.

**Comment by rosarior** [comment]
Fixed in 0eaf8cea79a859137b56aa88022006666e769838. Thanks!
